### PR TITLE
Cmr 9654 - remove old coll cache

### PR DESCRIPTION
**Pull request by jmaeng72**
Changes:
* Remove old collection for granules in-memory cache (which was replaced by external redis cache)
* Remove the newly created external redis cache: collection-for-gran-acls-by-provider-id-and-entry-title because it is not being used anywhere in the code

### CODE REVIEW

File: bootstrap-app/src/cmr/bootstrap/api/routes.clj [hunk 1]
```diff
@@ -133,8 +133,7 @@
               (= keyword-cache-name provider-cache/cache-key)
               (provider-cache/refresh-provider-cache request-context)
 
-              (or (= keyword-cache-name coll-for-gran-acls-caches/coll-by-concept-id-cache-key)
-                  (= keyword-cache-name coll-for-gran-acls-caches/coll-by-provider-id-and-entry-title-cache-key))
+              (= keyword-cache-name coll-for-gran-acls-caches/coll-by-concept-id-cache-key)
```
Comment (eereiter) [inline bootstrap-app/src/cmr/bootstrap/api/routes.clj:136]: What happens if someone requests the entry-title-cache-key refresh?

File: search-app/src/cmr/search/services/acls/collections_cache.clj [hunk 2]
```diff
@@ -1,151 +1,19 @@
...
-  (let [cache (hash-cache/context->cache context cache-key)
-        collections-map (fetch-collections-map context)]
-    (doseq [[coll-key coll-value] collections-map]
-      (hash-cache/set-value cache cache-key coll-key (clj-times->time-strs coll-value)))))
+    [cmr.common.hash-cache :as hash-cache]
+    [cmr.common.log :as log :refer (debug info warn error)]
+    [cmr.common-app.data.search.collection-for-gran-acls-caches :as coll-caches]))
 
 (defn get-collection-for-gran-acls
   ([context coll-concept-id]
```
Comment (eereiter) [inline search-app/src/cmr/search/services/acls/collections_cache.clj:9]: It looks like the get-collection-for-gran-acls context provider-d entry-title is not being used.

In that case on new line 9 remove the extra ( and one at the end of this function as you don't need them anymore.

**Review by eereiter — APPROVED** [review]
I guess you don't need the second cache anymore.  that is good.

File: common-app-lib/src/cmr/common_app/data/search/collection_for_gran_acls_caches.clj [hunk 3]
```diff
@@ -109,43 +91,26 @@
...
-      (hash-cache/set-value coll-by-concept-id-cache
-                            coll-by-concept-id-cache-key
-                            (:concept-id collection-found)
-                            (clj-times->time-strs collection-found))
-      (hash-cache/set-value coll-by-provider-id-and-entry-title-cache
-                            coll-by-provider-id-and-entry-title-cache-key
-                            (str (:provider-id collection-found) (:EntryTitle collection-found))
-                            (clj-times->time-strs collection-found)))
-    collection-found))
+         " coll-by-concept-id-cache Cache Size: " (hash-cache/cache-size coll-by-concept-id-cache coll-by-concept-id-cache-key) " bytes"))))
```
Comment (jceaser) [inline common-app-lib/src/cmr/common_app/data/search/collection_for_gran_acls_caches.clj:101]: format would make this easy to read

File: common-app-lib/src/cmr/common_app/data/search/collection_for_gran_acls_caches.clj [hunk 4]
```diff
@@ -1,10 +1,9 @@
 (ns cmr.common-app.data.search.collection-for-gran-acls-caches
-"Defines common functions and defs a cache for search collections.
- This namespace defines two separate caches that are tightly coupled and will be created and updated/refreshed at the same time:
- - coll-by-concept-id-cache = collection-for-gran-acls-by-concept-id-cache
- - coll-by-entry-title-cache = collection-for-gran-acls-by-provider-id-and-entry-title-cache"
+  "Defines common functions and defs a cache for search collections.
```
Comment (jceaser) [inline common-app-lib/src/cmr/common_app/data/search/collection_for_gran_acls_caches.clj:2]: space is correct?

**Review by jceaser — APPROVED** [review]

File: bootstrap-app/src/cmr/bootstrap/api/routes.clj (hunk 1 above)
Comment (jmaeng72) [reply bootstrap-app/src/cmr/bootstrap/api/routes.clj:136]: I've removed the cache entirely, so it should not be called anywhere

Comment (jmaeng72) [reply bootstrap-app/src/cmr/bootstrap/api/routes.clj:136]: that path won't exist anymore and neither will that cache

File: common-app-lib/src/cmr/common_app/data/search/collection_for_gran_acls_caches.clj (hunk 4 above)
Comment (jmaeng72) [reply common-app-lib/src/cmr/common_app/data/search/collection_for_gran_acls_caches.clj:2]: Yes. Two spaces.

File: search-app/src/cmr/search/services/acls/collections_cache.clj (hunk 2 above)
Comment (jmaeng72) [reply search-app/src/cmr/search/services/acls/collections_cache.clj:9]: will do

**Pull request merged**